### PR TITLE
Update GameServerSet scheduling when Fleet scheduling is changed.

### PR DESCRIPTION
**Pull request by pooneh-m**
Fixing issue #495

### CODE REVIEW

**Comment by jkowalski** [comment]
very nice, can you also add e2e test?

**Comment by pooneh-m** [comment]
> very nice, can you also add e2e test?

Done! :)

**Comment by markmandel** [comment]
This is technically a bug, and could be merged during feature freeze, but I'm thinking we leave this until next release, just to give the current code some time to be tested by our users.

Happy to take dissenting opinions.

File: test/e2e/framework/framework.go [hunk 1]
```diff
@@ -174,30 +174,34 @@ func FleetReadyCount(amount int32) func(fleet *v1alpha1.Fleet) bool {
 	}
 }
 
-// WaitForFleetGameServersCondition wait for all GameServers for a given
-// fleet to match the spec.replicas and match a a condition
-func (f *Framework) WaitForFleetGameServersCondition(flt *v1alpha1.Fleet, cond func(server v1alpha1.GameServer) bool) error {
+// WaitForFleetGameServersCondition waits for all GameServers for a given fleet to match
+// a condition spacified by a callback.
```
Comment (aLekSer) [inline test/e2e/framework/framework.go]: small typo
```suggestion
// a condition specified by a callback.
```

Comment (markmandel) [reply test/e2e/framework/framework.go]: Nice catch. Outside of this, this looks good to go :+1:

Comment (pooneh-m) [reply test/e2e/framework/framework.go]: Thanks! Fixed it.

**Pull request merged**